src/server/service/leaderboard.ts

```diff
@@ -0,0 +1,6 @@
+import { type BBPgTransaction } from "~/server/db/index.js";
+import { updateRanks } from "~/server/db/rank.js";
+
+const calculate = (tx?: BBPgTransaction) => updateRanks(tx);
```

Comment:
Why just renaming?

Reply:
This is the leaderboard service, where I want to abstract all the functionality related to the leaderboard, including interaction with the program. This way whenever you want to interface with the leaderboard, you know that you can use this single service instead of looking in n different files.

Reply:
When I call it at a higher level it makes more sense to call it like `leaderboard.calculate` instead of explicitly saying that I am updating the rank table.